nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-2-6-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/PublishKafkaRecord_2_6.java

```diff
@@ -124,6 +125,14 @@
         "Interprets the <Partition> property as Expression Language that will be evaluated against each FlowFile. This Expression will be evaluated once against the FlowFile, " +
             "so all Records in a given FlowFile will go to the same partition.");
 
+    static final AllowableValue MESSAGE_PROPERTY_KEY_RESOLVER_SELECTION = new AllowableValue("MessageValuePropertyResolver",
+            "MessageValuePropertyResolver", "The <Message Key Field> property will be interpreted as the name of a simple, " +
+            "non-nested property of the message value. This reflects the previously existing behaviour.");
+    static final AllowableValue RECORD_PATH_KEY_RESOLVER_SELECTION = new AllowableValue("RecordPathResolver",
```

Comment:
It's probably worth noting here that if the RecordPath matches more than one element, the first one will be chosen. And that the value will be simply converted to a String and is expected to be a primitive value, not an embedded record. I can definitely envision someone expecting to use this with a RecordPath that points to a complex JSON object or a complex Avro record, for instance.